lib/utils/KalturaCouchbaseConnector.js

```diff
@@ -0,0 +1,379 @@
+let util = require('util');
+let Promise = require("bluebird");
+
+let base = require('../KalturaBase');
+
+let mock = false;
+let couchbase = require('couchbase');
+if (mock)
+{
+    couchbase = couchbase.Mock;
+}
+
+class KalturaCouchbaseConnector {
+
+    constructor(){
+        this._config = KalturaConfig.config.couchbase;
+        this._clusterName = this._config.clusterName;
+        this._bucketName = this._config.bucketName;
+        this._cluster = new couchbase.Cluster(this._clusterName);
+        this._bucket = this._cluster.openBucket(this._bucketName);
+        this._dataVersion = 0;
+
+        if (this._config.dataVersion)
+            this._dataVersion = parseInt(this._config.dataVersion);
+
+        process.on('unhandledRejection', (reason) => {
+            KalturaLogger.log(reason);
+        });
+    }
+
+    static getInstance()
+    {
+        if (typeof KalturaCouchbaseConnector.instance == 'undefined')
+        {
+            KalturaCouchbaseConnector.instance = new KalturaCouchbaseConnector();
+        }
+        return KalturaCouchbaseConnector.instance;
+    }
+
+    getDataVersion()
+    {
+        return this._dataVersion;
+    }
+
+    getStack()
+    {
+        return new Error();
+    }
+
+    errorHandling(reject, err)
+    {
+        let stackSource = this.getStack();
+        var errMessage = 'Cache.get :' + err;
+        if(reject){
+            reject(errMessage);
+        }
+        else{
+            KalturaLogger.error(errMessage + "\n" + stackSource.stack, stackSource);
+        }
+    }
+
+    get(key, callback, errorCallback, is_encrypted)
+    {
+        let This = this;
+        let stackSource = this.getStack();
+        KalturaLogger.debug('Couchbase.get [' + key + ']...', stackSource);

```

Comment:
Instead of adding stackSource - call KalturaLogger.err , that will add stack trace by default.